File: 0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> twoSum; 
        map<int, int> mp;
        for (int i = 0; i < nums.size(); i++) {
            int n = nums[i];
            int k = target - n;
            if (mp.find(k) != mp.end()) {
                twoSum.push_back(mp[k]); 
                twoSum.push_back(i);     
                return twoSum;
            }
            mp[n] = i;
        }
        return twoSum;
    }
};

File: 0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] twoSum = new int[2];
        for (int i = 0; i < nums.length; i++) {
            for (int j = i+1; j < nums.length; j++) {
                // if (i != j) {
                    int add = nums[i] + nums[j];
                    if (add == target) {
                        twoSum[0] = i;
                        twoSum[1] = j;
                        return twoSum;
                    }
            }
        }
        return new int[0];
    }
}

